lib/enigma.rb

```diff
@@ -14,11 +14,67 @@
 # > e.crack(output) # Date is optional, use today's date
 # => "this is so secret ..end.."
 #
-# def encrypt
-# end
-#
-# def decrypt
-# end
-#
-# def crack
-# end
+require_relative 'key_generator'
+require_relative 'offset_calculator'
+
+
+class Enigma
+  attr_accessor :key, :oc
+
+  def initialize(oc = OffsetCalculator.new)
+    @counter = 0
+    @characters = ("a".."z").to_a
+    @oc = oc
+  end
+
+  def hash_zip(rotation = oc.offset)
+    rotated_characters = @characters.rotate(rotation)
+    Hash[@characters.zip(rotated_characters)]
+  end
+
+  def encrypt_letter(letter, rotation)
+    cipher_for_rotation = hash_zip(rotation)
+    cipher_for_rotation[letter]
+  end
+
+  def encrypt(message, key = @key, date = @date)
+    offset = OffsetCalculator.new(key.to_i, date)
+    hash_zip(rotation = oc.offset)
+    message.chomp
+    encrypted = message
+    length = message.length
+    def find_a_rot
+      counter = 0
+      while counter < message.length
```

Comment:
These methods are very similar, is there away to make this generalized so you don't have to copy paste? What happens if you want to change something? You'll have to change it in four places!